### PR TITLE
Upgrade to 0.12

### PR DESCRIPTION
**Pull request by hugolesta**
This PR is for upgrade from 0.11 to 0.12 terraform version.

Features:

- main.tf better organized
- New syntax v0.12
- New setup.cf file
- New details in Readme.cf file
- Added new Backwards Incompatibilities notes
- Added new version.tf file for terraform.v0.12

### CODE REVIEW

**Comment by hugolesta** [comment]
@fdiazydiaz, @jverhoeks  Could you please help me with the review of this PR? Thanks!

File: README.md [hunk 1]
```diff
@@ -1,28 +1,87 @@
-# tf-aws-iam-cloudability
```
Comment (franciscodiazydiaz) [inline README.md:1]: This first line represents the name of the module I would keep it also with the description.

File: README.md [hunk 2]
```diff
@@ -1,28 +1,87 @@
...
+| automation_policy_name | Name of automation resource policy | string | `CloudabilityAutomationPolicy` | no |
+| enabled | Enable Policies and role | string | `true` | no |
+| external_id | The external identifier to use, given to you by Cloudability | string | `xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx` | no |
+| monitor_policy_name | Name of monitor resource policy | string | `CloudabilityMonitorResourcesPolicy` | no |
+| policy_path | Path in IAM for policies | string | `/` | no |
+| role_name | Name of Cloudability Role | string | `CloudabilityRole` | no |
+| role_path | Path in IAM for role | string | `/` | no |
+| trusted_account_id | Trusted Account from Cloudability | string | `xxxxxxxx` | no |
 
-## License
```
Comment (franciscodiazydiaz) [inline README.md:37]: As this is OS, the License should be specified

File: README.md [hunk 3]
```diff
@@ -1,28 +1,87 @@
...
+When it already exists terraform gives this error:
+```
+* aws_iam_role.cloudability_role: Error creating IAM Role CloudabilityRole: EntityAlreadyExists: Role with name CloudabilityRole already exists.
+	status code: 409, request id: xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx
+```
+
+### Import The role:
+
+To import the resource Terraform has the following command available:
+`terraform  import  module.cloudability.aws_iam_role.cloudability_role  CloudabilityRole`
```
Comment (franciscodiazydiaz) [inline README.md]: Probably is better to use the code snippet format than a single `.
Same for the command described bellow.

File: setup.cf [hunk 4]
```diff
@@ -0,0 +1,150 @@
+{
```
Comment (franciscodiazydiaz) [inline setup.cf]: Is this file included intentionally?

File: README.md [hunk 5]
```diff
@@ -1,28 +1,87 @@
...
+
+The wrapper script has this option to include the assume role and run import for this account
+`./runall.nl -b <accountid-description> -t module.cloudability.aws_iam_role.cloudability_role  -v CloudabilityRole`
```
Comment (franciscodiazydiaz) [inline README.md]: Where this command comes from?

File: README.md [hunk 6]
```diff
@@ -1,28 +1,87 @@
...
+path                  = /
+unique_id             = xxxxxx
+```
+
+```
+cloudability_external_id = "<UUID given by cloudability or dummy"
+```
+_there needs to be a string in the variable or the module isn't loaded_
+
+
```
Comment (franciscodiazydiaz) [inline README.md]: Some unnecessary spaces.

**Review by franciscodiazydiaz — COMMENTED** [review]
Added some comments inline. Thanks for the improvements!

File: README.md (hunk 1 above)
Comment (hugolesta) [reply README.md:1]: Thanks for the update, done.

File: README.md (hunk 3 above)
Comment (hugolesta) [reply README.md]: Done!

File: README.md (hunk 5 above)
Comment (hugolesta) [reply README.md]: It seems, this code is not necessary, It has heritance from other repo, done!

File: README.md (hunk 6 above)
Comment (hugolesta) [reply README.md]: Some unnecessary spaces have been deleted.

File: README.md (hunk 2 above)
Comment (hugolesta) [reply README.md:37]: I wrote the previous License. It was deleted by mistake

File: setup.cf (hunk 4 above)
Comment (hugolesta) [reply setup.cf]: The file has been deleted.

File: README.md (hunk 5 above)
Comment (hugolesta) [reply README.md]: Apparently that command is not necessary anymore

**Review by nahueloyha — APPROVED** [review]
LGTM! But please check if there is sth to add from the previous @jverhoeks's PR: https://github.com/edrans/tf-aws-iam-cloudability/pull/1/ :)

**Comment by jverhoeks** [comment]
Thanks, synced and merged

**Pull request merged**